src/langchain_google_alloydb_pg/alloydb_chat_message_history.py

```diff
@@ -15,35 +15,86 @@
 from __future__ import annotations
 
 import json
-from typing import List
+from typing import List, Sequence
 
 from langchain_core.chat_history import BaseChatMessageHistory
-from langchain_core.messages import BaseMessage, message_to_dict, messages_from_dict
+from langchain_core.messages import BaseMessage, messages_from_dict
 
 from .alloydb_engine import AlloyDBEngine
 
 
+async def _aget_messages(
+    engine: AlloyDBEngine, session_id: str, table_name: str
+) -> List[BaseMessage]:
+    """Retrieve the messages from AlloyDB"""
+    query = f"""SELECT data, type FROM "{table_name}" WHERE session_id = :session_id ORDER BY id;"""
+    results = await engine._afetch(query, {"session_id": session_id})
+    if not results:
+        return []
+
+    items = [{"data": result["data"], "type": result["type"]} for result in results]
+    messages = messages_from_dict(items)
+    return messages
+
+
 class AlloyDBChatMessageHistory(BaseChatMessageHistory):
     """Chat message history stored in an AlloyDB."""
 
-    def __init__(self, engine: AlloyDBEngine, session_id: str, table_name: str):
+    __create_key = object()
+
+    def __init__(
+        self,
+        key,
+        engine: AlloyDBEngine,
+        session_id: str,
+        table_name: str,
+        messages: List[BaseMessage],
+    ):
+        if key != AlloyDBChatMessageHistory.__create_key:
+            raise Exception(
+                "Only create class through 'create' or 'create_sync' methods!"
+            )
         self.engine = engine
         self.session_id = session_id
         self.table_name = table_name
+        self.messages = messages
```

Comment:
If `.messages` no longer reads from the database and instead uses this list is there a chance the DB and this list could get out of sync somehow?

Reply:
I was a little worried about that too but this also means we don't have to query the database each time we read messages

Reply:
This also concerns me... are we only writing to the database then?